iroh/src/sync_engine/live.rs

```diff
@@ -546,35 +510,94 @@ impl<B: iroh_bytes::store::Store> LiveActor<B> {
                     debug!(%e, "failed to register peer for document")
                 }
 
-                SyncState::Finished
+                // broadcast a sync report to our neighbors, but only if we received new entries.
+                if details.outcome.num_recv > 0 {
+                    info!("broadcast sync report to neighbors");
+                    let report = SyncReport {
+                        namespace,
+                        heads: details.outcome.heads_received.clone(),
+                    };
+                    self.broadcast_neighbors(namespace, &Op::SyncReport(report))
+                        .await;
+                }
             }
-            Err(ref err) => {
-                warn!(?origin, ?err, "sync failed");
+        };
 
-                SyncState::Failed
-            }
+        let result_for_event = match &result {
+            Ok(_) => Ok(()),
+            Err(err) => Err(err.to_string()),
         };
 
-        self.set_sync_state(namespace, peer, state);
+        let Some((started, resync)) = self.state.finish(&namespace, peer, &origin, result) else {
+            return;
+        };
 
         let ev = SyncEvent {
-            namespace,
             peer,
             origin,
-            result: result
-                .as_ref()
-                .map(|_| ())
-                .map_err(|err| format!("{err:?}")),
+            result: result_for_event,
             finished: SystemTime::now(),
+            started,
         };
         self.subscribers
             .send(&namespace, Event::SyncFinished(ev))
             .await;
-        Ok(())
+
+        if resync {
+            self.sync_with_peer(namespace, peer, SyncReason::Resync);
+        }
     }
 
-    fn is_syncing(&self, namespace: &NamespaceId) -> bool {
-        self.is_syncing.contains(namespace)
+    async fn broadcast_neighbors(&self, namespace: NamespaceId, op: &Op) {
+        if !self.state.is_syncing(&namespace) {
+            return;
+        }
+
+        let msg = match postcard::to_stdvec(op) {
+            Ok(msg) => msg,
+            Err(err) => {
+                error!(?err, ?op, "Failed to serialize message:");
+                return;
+            }
+        };
+        // TODO: We should debounce and merge these neighbor announcements likely.
```

Comment:
yeah, definitely, if you have a fast producer this will trigger a lot of traffic

Reply:
We should. Let's do it in a followup though?

Reply:
sounds good